pkg/api/validation/validation.go

```diff
@@ -97,6 +97,10 @@ func ValidateAnnotations(annotations map[string]string, fldPath *field.Path) fie
 	if totalSize > (int64)(totalAnnotationSizeLimitB) {
 		allErrs = append(allErrs, field.TooLong(fldPath, "", totalAnnotationSizeLimitB))
 	}
+
+	if annotations[api.AffinityAnnotationKey] != "" {
```

Comment:
Yeah, but never mind, what you have is fine (and I see you did it the same way in GetAffinityFromPodAnnotations()). So no need to change anything.